Fixes from the evaluation

mmagician opened the pull request:
- fix unit test names
- add TODOs and let unwrap
- subscription record is always new, reuse it
- refactor set_subscrypt_pass, add TODO
- add more todos
- refactor check_auth and add a test

Commit:
fix unit test names

Commit:
add TODOs and let unwrap

Commit:
subscription record is always new, reuse it

Commit:
refactor set_subscrypt_pass, add TODO

Commit:
add more todos

Commit:
refactor check_auth and add a test

SaberDoTcodeR committed:
max_refund_percent_policy var name refactor

SaberDoTcodeR committed:
mod.rs max_refund_permille_policy compatibility

HadiEsna committed:
resolve TODOs

HadiEsna committed:
Merge branch 'rust-unwrap-fixes' of https://github.com/mmagician/SubsCrypt-ink into rust-unwrap-fixes

HadiEsna committed:
revert the unresolved TODO

SaberDoTcodeR committed:
set_subscrypt_pass test added

HadiEsna commented on src/lib.rs:
```diff
@@ -494,32 +491,31 @@ pub mod subscrypt {
         /// Examples in `withdraw_works` and `withdraw_works2` in `tests/test.rs`
         #[ink(message)]
         pub fn withdraw(&mut self) -> u128 {
+            // TODO match & unwrap here
```
Sorry  @mmagician , I think the current implementation is simple and clean enough.
I'm not sure that I understood your idea to enhance the current code.

mmagician replied:
Makes sense

SaberDoTcodeR committed:
test function names

test function names are now more explanatory
some asserts without panic message now have message

SaberDoTcodeR committed:
compiled version in tag v0.1

https://github.com/oxydev/SubsCrypt-ink/releases/tag/0.1

mmagician commented:
The changes look good, good to go on my side
